### PR TITLE
550: all: enable checkstyle for tests and fix style errors

### LINKED ISSUE #550: Check targets appear to be broken when running locally
When running `./mvnw checkstyle:check` task reports success while it fails on CI because of unused import.

Also when running `./mvnw license:check` following error message is printed:

```text
[ERROR] Failed to execute goal com.mycila:license-maven-plugin:3.0:check (default-cli) on project jetcd-parent: Execution default-cli of goal com.mycila:license-maven-plugin:3.0:check failed: Cannot read header document ${jetcd.project.root}/etc/license.txt. Cause: Resource ${jetcd.project.root}/etc/license.txt not found in file system, classpath or URL: no protocol: ${jetcd.project.root}/etc/license.txt -> [Help 1]
```

### PR DESCRIPTION
**Pull request by yb172**
For better consistency enable checkstyle for tests and reformat tests to comply with rules.

Fixes #550

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -708,7 +710,7 @@
                 <artifactId>license-maven-plugin</artifactId>
                 <version>${license-maven-plugin.version}</version>
                 <configuration>
-                    <header>${jetcd.project.root}/etc/license.txt</header>
+                    <header>${session.executionRootDirectory}/etc/license.txt</header>
```
Comment (yb172) [inline pom.xml]: @lburgazzoli does that look ok to you? I'm not an expert in maven, but with this change `./mvnw license:check` works locally for me.

**Comment by yb172** [comment]
@fanminshi @lburgazzoli Please take a look

File: pom.xml (hunk 1 above)
Comment (lburgazzoli) [reply pom.xml]: it does not look correct as that variable points to the path where the maven command was executed so if I go to a submodule, then the license check would not work.

That is the reason why the property jetcd.project.root exists, we should investigate why it didn’t get resolved.

Comment (yb172) [reply pom.xml]: All right, thank you for explanation. I would revert it back then.

File: jetcd-core/src/test/java/io/etcd/jetcd/WatchUnitTest.java [hunk 2]
```diff
@@ -321,28 +320,28 @@ public void testWatcherCreateOnCancellationWithNoReason() throws InterruptedExce
...
 
     try (Watch.Watcher watcher = watchClient.watch(KEY, listener)) {
       WatchResponse createdResponse = createWatchResponse(0);
       responseObserverRef.get().onNext(createdResponse);
 
       WatchResponse canceledResponse = WatchResponse
-              .newBuilder()
-              .setCanceled(true)
-              .build();
+          .newBuilder()
```
Comment (fanminshi) [inline jetcd-core/src/test/java/io/etcd/jetcd/WatchUnitTest.java]: is this auto formatted? not sure why it has 4 spaces as the indent.

File: jetcd-core/src/test/java/io/etcd/jetcd/WatchUnitTest.java [hunk 3]
```diff
@@ -351,29 +350,29 @@ public void testWatcherCreateOnCancellationWithReason() throws InterruptedExcept
     CountDownLatch latch = new CountDownLatch(1);
     AtomicReference<Throwable> ref = new AtomicReference<>();
     Watch.Listener listener = Watch.listener(
-      r -> {},
-      t -> {
-        ref.set(t);
-        latch.countDown();
-      }
+        r -> { },
```
Comment (fanminshi) [inline jetcd-core/src/test/java/io/etcd/jetcd/WatchUnitTest.java]: here too.

**Comment by fanminshi** [comment]
LGTM if all the changes are according to checkstyle and google java style sheet.

File: jetcd-core/src/test/java/io/etcd/jetcd/WatchUnitTest.java (hunk 2 above)
Comment (yb172) [reply jetcd-core/src/test/java/io/etcd/jetcd/WatchUnitTest.java]: It's formatted according to styleguide. Weird thing is that linter complains if indentation is less that what it should be, but not when it's more.

Comment (fanminshi) [reply jetcd-core/src/test/java/io/etcd/jetcd/WatchUnitTest.java]: i see.

**Pull request merged**